### DISCUSSION TITLE
What is best practice on registers a custom block type in the Gutenberg with ACF Pro Blocks in Understrap with custom css styling?

### DISCUSSION THREAD

**Discussion by isaklafleur** [holger1411/understrap#1512] [Q&A]
https://www.advancedcustomfields.com/resources/acf_register_block_type/

```
add_action('acf/init', 'my_acf_blocks_init');
function my_acf_blocks_init() {

    // Check function exists.
    if( function_exists('acf_register_block_type') ) {

        // Register a testimonial block.
        acf_register_block_type(array(
            'name'              => 'testimonial',
            'title'             => __('Testimonial'),
            'description'       => __('A custom testimonial block.'),
            'render_template'   => 'template-parts/blocks/testimonial/testimonial.php',
            'category'          => 'formatting',
            'enqueue_style' => get_template_directory_uri() . '/css/child-theme.min.css',
        ));
    }
}
```

Above is an example code with custom css for the Block style. In understrap-child all scss is compiled to css, so should I always refer to child-theme.min.css on all my custom blocks or do you recommend to work in another way?

![image](https://user-images.githubusercontent.com/10033653/142183380-df59a078-8ab5-4e87-b7d2-b8e4d0cd34f6.png)

![image](https://user-images.githubusercontent.com/10033653/142182786-92510b94-4a8a-4afe-8dde-0276cff18d23.png)

**Comment by bacoords** [comment]
Hi- 

You have the option to load small and separate CSS files for your each of custom ACF blocks. The benefit here is they'll inject them into the front-end when your block is used AND the back-end when your block is showing in Gutenberg. The downside is they are separate files for each block and if you're not using a plugin to combine CSS files on your live site, you end up with a lot of separate files.

The other option is you can just add all the block-related CSS to your child theme and take advantage of the SCSS tooling in Understrap. For example, we have a [_blocks.scss](https://github.com/understrap/understrap/blob/main/src/sass/theme/_blocks.scss) file that is loaded in the front-end and back-end automatically and includes block styling. The benefit of using this method is you can take advantage of some of the bootstrap and sass features like @extend or @include.

**Reply by isaklafleur** [reply]
Hi Brian,

Thank you for the quick reply. We are trying to use almost only bootstrap 5, but this is great tips if we need to do any custom CSS in our blocks.

I have done a block that is taking up vh-100 (using bootstrap classes), in Gutenberg editor this block is not shown with vh-100. Any clue how to fix this?

**Reply by isaklafleur** [reply]
Btw, we are using your bs5 development version. Really like it! I have done some updates in the package.json file, if you like I can share those as pull request. Please let me know.

**Reply by bacoords** [reply]
We don't import ALL of the bootstrap classes into the backend editor- but you can see here what we do import, in case you want to bring in additional parts of bootstrap: https://github.com/understrap/understrap/blob/develop/src/sass/custom-editor-style.scss

Sure please share a PR or link to your repo! We're always looking for more contributions.
